sklearn/utils/extmath.py

```diff
@@ -174,6 +174,24 @@ def density(w, **kwargs):
     return d
 
 
+def _csr_to_array_fallback(X):
+    """ This provides the equivalent functionality to
+       csr_matrix.toarray()
+    and provides a fix for scipy versions <1.0.0 where
+    the above command fails for arrays with more than 2**31
+    elements.
+    """
+    if X.format != 'csr':
+        raise ValueError('Only CSR sparse arrays are supported')
+    row_indices = np.zeros(len(X.indices), dtype='int32')
+    indptr = X.indptr
+    for idx in range(len(indptr) - 1):
+        row_indices[indptr[idx]:indptr[idx+1]] = idx
+    Y = np.zeros(X.shape, dtype=X.dtype)
+    Y[row_indices, X.indices] = X.data
+    return Y
```

Comment:
I would rather put that function into `sklearn/utils/fixes.py` which is dedicated to numpy and scipy backports.